### ISSUE TITLE
Pre-training and Data Augmentation

### ISSUE THREAD

**Issue by tonmoyborah** [nwojke/cosine_metric_learning#16]
Hi Nwojke,
I had a few doubts I would like to clear. Since this is not pre-trained on Imagenet and you report that performance can increase with imagenet pre training, how should I go about it if I want to pre-train it? Will the improvement be significant in your opinion?
Also is test time augmentation done while reporting the results of this paper?
Again, since the image names carry vital info like tracklet id, label and camera, how can one do train data augmentation and still somehow take care of the image names? I presume these info are used during training somehow and augmenting will only disturb it. What are your thoughts on it?

**Comment by nwojke** [comment]
Hi tonmoyborah,

sorry for the late reply, I hope my answer it still helps.

* For ImageNet pretraining you could either write your own training script around the network architecture code provided in this repository or you try to integrate a pretrained network into this repository. I am not sure what is the better fit in your case.
* We didn't do test time augmentation.
* It is best to do data augmentation *on the fly*, i.e., load raw images and their labels from disk, then manipulate the image data according to data augmentation, then place it in the training queue. This way, you won't have any issues with filenames and don't generate excessive amounts of data on your hard drive.

**Comment by nwojke** [comment]
I am closing this issue due to inactivity. Re-open if the problem is not solved.
